state/runtime/evm/instructions.go

```diff
@@ -1173,7 +1182,7 @@ func opCall(op OpCode) instruction {
 		}
 
 		if result.Succeeded() || result.Reverted() {
-			if len(result.ReturnValue) != 0 {
+			if len(result.ReturnValue) != 0 && offset+size <= uint64(len(c.memory)) {
```

Comment:
What happens if the `offset+size` is larger than the length of memory? Should we exit if this is true, like we do in `opReturnDataCopy`?

Reply:
offset+size will be  larger than the length of memory only if the size is 0 because only in that case memory would not be allocated. according to the behavior in eht-go(code sample given in the audit) if size is not greater then 0 then nothing should be done. But probably would be better if I put condition size>0 as it is done in other places (opExtCodeCopy, opCallDataCopy, opCodeCopy, get2) when memory is accessed after allocate is called. 
I can change this condition to: if len(result.ReturnValue) != 0 && size>0  to be aligned with other places